data generator in feed demo returns error causes e2e test failure

### Which jobs are flaking?

engine_ghpr_integration_test

### Which test(s) are flaking?

e2e_test.go: TestSubmitTest

### Jenkins logs or GitHub Actions link

https://ci2.pingcap.net/blue/organizations/jenkins/engine_ghpr_integration_test/detail/engine_ghpr_integration_test/591/pipeline

detail logs: 
[log-e2e_basic-dm_full_mode.tar.gz](https://github.com/pingcap/tiflow/files/9583326/log-e2e_basic-dm_full_mode.tar.gz)

```go

[2022-09-16T09:40:27.115Z] === RUN   TestSubmitTest
[2022-09-16T09:40:27.115Z] connect demo 127.0.0.1:1234
[2022-09-16T09:40:29.008Z]     e2e_test.go:91: 
[2022-09-16T09:40:29.008Z]         	Error Trace:	/home/jenkins/agent/workspace/engine_ghpr_integration_test/go/src/github.com/pingcap/tiflow/engine/test/e2e/e2e_test.go:91
[2022-09-16T09:40:29.008Z]         	Error:      	Expected nil, but got: &status.Error{s:(*status.Status)(0xc000b321e8)}
[2022-09-16T09:40:29.008Z]         	Test:       	TestSubmitTest
[2022-09-16T09:40:29.008Z] --- FAIL: TestSubmitTest (5.00s)
[2022-09-16T09:40:29.008Z] FAIL
[2022-09-16T09:40:29.008Z] FAIL	github.com/pingcap/tiflow/engine/test/e2e	5.110s
[2022-09-16T09:40:29.008Z] FAIL
```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

--- a/engine/test/e2e/docker.json
+++ b/engine/test/e2e/docker.json
@@ -2,7 +2,7 @@
     "demo_address": ["127.0.0.1:1234"],
     "demo_host": ["demo-server:1234"],
     "master_address_list": ["127.0.0.1:10245", "127.0.0.1:10246", "127.0.0.1:10247"],
-    "demo_record_num": 5000000,
+    "demo_record_num": 50000,
     "job_num": 2,
     "file_num": 5
 }

--- a/engine/test/e2e/e2e_test.go
+++ b/engine/test/e2e/e2e_test.go
@@ -81,14 +81,14 @@ func TestSubmitTest(t *testing.T) {
 		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 		defer cancel()
 		democlient, err := NewDemoClient(ctx, demoAddr)
-		require.Nil(t, err)
+		require.NoError(t, err)
 		fmt.Println("connect demo " + demoAddr)
 
 		resp, err := democlient.client.GenerateData(ctx, &pb.GenerateDataRequest{
 			FileNum:   int32(config.FileNum),
 			RecordNum: int32(config.RecordNum),
 		})
-		require.Nil(t, err)
+		require.NoError(t, err)
 		require.Empty(t, resp.ErrMsg)
 	}
 
@@ -131,12 +131,12 @@ func testSubmitTest(t *testing.T, cfg *cvs.Config, config *Config, demoAddr stri
 	ctx := context.Background()
 	fmt.Printf("connect demo\n")
 	democlient, err := NewDemoClient(ctx, demoAddr)
-	require.Nil(t, err)
+	require.NoError(t, err)
 	fmt.Printf("connect clients\n")
 
 	for {
 		resp, err := democlient.client.IsReady(ctx, &pb.IsReadyRequest{})
-		require.Nil(t, err)
+		require.NoError(t, err)
 		if resp.Ready {
 			break
 		}
@@ -144,7 +144,7 @@ func testSubmitTest(t *testing.T, cfg *cvs.Config, config *Config, demoAddr stri
 	}
 
 	configBytes, err := json.Marshal(cfg)
-	require.Nil(t, err)
+	require.NoError(t, err)
 
 	<-flowControl
 	fmt.Printf("test is ready\n")
@@ -176,6 +176,6 @@ func testSubmitTest(t *testing.T, cfg *cvs.Config, config *Config, demoAddr stri
 	demoResp, err := democlient.client.CheckDir(ctx, &pb.CheckDirRequest{
 		Dir: cfg.DstDir,
 	})
-	require.Nil(t, err, jobID)
+	require.NoError(t, err)
 	require.Empty(t, demoResp.ErrMsg, demoResp.ErrFileIdx)
 }

